src/vs/platform/telemetry/common/telemetryLogAppender.ts

```diff
@@ -23,7 +27,14 @@ export class TelemetryLogAppender extends Disposable implements ITelemetryAppend
 		if (logger) {
 			this.logger = this._register(logger);
 		} else {
-			this.logger = this._register(loggerService.createLogger(environmentService.telemetryLogResource));
+			// Not a perfect check, but a nice way to indicate if we only have logging enabled for debug purposes and nothing is actually being sent
+			const justLoggingAndNotSending = isLoggingOnly(productService, environmentService);
+			const logSuffix = justLoggingAndNotSending ? ' (Not Sent)' : '';
+			const telemetryLogResource = environmentService.telemetryLogResource;
+			const isVisible = () => supportsTelemetry(productService, environmentService) && logService.getLevel() === LogLevel.Trace;
+			this.logger = this._register(loggerService.createLogger(telemetryLogResource, { id: telemetryLogChannelId, name: localize('telemetryLog', "Telemetry{0}", logSuffix), hidden: !isVisible() }));
+			this._register(logService.onDidChangeLogLevel(() => loggerService.setVisibility(telemetryLogResource, isVisible())));
+
```

Comment:
CC @lramos15 - You do not need to create output channel now. You can include all necessary info while creating the logger and system will create it whenever you want. For eg., here you can change the visibility of the logger and it will allow system to create/remove the channel.